docs/docs/gatsby-provision.md

```diff
@@ -0,0 +1,44 @@
+
+This document will give an overview of the convention requirements, provide links to existing `gatsby-provision-*` packages as well as a suggest guidelines for those looking to author their own Gatsby Provision package.
+
+## Gatsby Provision Convention Overview
+
+The goal of the `gatsby-provision` convention is to provide a standardized manner with which Gatsby sites can populate content into an accompanying data source. A typical use case would be if you're building a Gatsby Starter, or a boilerplate or template site for your own re-use.
+
+The generated content could be anything from markdown files, to creating a new database, to populating a CMS — but usually it will involve interacting with third parties which require some means of authentication.
+
+Another goal of this standardization is to enable deployment of a Gatsby site in Gatsby Cloud without requiring a pre-requisite step of setting up the data source(s).
+
+In short, in order to implement the `gatsby-provision` convention, your site should:
+
+- Include a `gatsby-provision-*` package as a `devDependency`
+- Have a npm script named `gatsby-provision` in your site's `package.json` file which invokes the `gatsby-provision-*` package with any required arguments.
+
+## Existing Gatsby Provision packages
+
+Currently there exists the following packages that can be used to provision content in their respective systems / vendors:
+
+- `gatsby-provision-contentful` - [README](https://github.com/gatsbyjs/gatsby-provision-contentful)
+- `gatsby-provision-sanity` - [README](https://github.com/gatsbyjs/gatsby-provision-sanity)
+
+See the associated README link for usage details for each package.
+
+## Building your own Gatsby Provision package
+
+Given the context and examples above, if you're reading this section, you probably already have an idea of what logic your Gatsby Provision script should contain. However, there are some common aspects to what a `gatsby-provision` script should be designed to adhere to:
+
+- Be able to execute the CLI package by name — if you are unfamiliar, here is a good [article on how to build a CLI with Node](https://dev.to/rushankhan1/build-a-cli-with-node-js-4jbi)
+- Work both locally and in Gatsby Cloud, discerning between the two contexts via an environment variables named `GATSBY_CLOUD`. When the script is ran in Gatsby Cloud, the following statement will resolve as true: `process.env.GATSBY_CLOUD === "true"`
+- Utilize a combination of bash arguments and environment variables to source dynamic values the script requires
+- In the local context, it should write out or amend a `.env` / `.env.development` / `.env.production` file(s) with any values (API tokens, CMS namespaces, etc.) that will be required environment variables for the site to build
+- In the local context, for a better DX consider using an interactive CLI tool such as `inquirer` - [inquirer on npm](https://www.npmjs.com/package/inquirer)
```

Comment:
Is it ok to say DX instead of developer experience? I don't actually know, just wondering and you're the actual developer :)

Reply:
Nah, good point on avoiding acronyms. Doesn't hurt to spell it out!